qtpy/tests/test_qtcore.py

```diff
@@ -16,34 +16,41 @@
 )
 from qtpy.tests.utils import not_using_conda
 
+NOW = datetime.now()
+# Make integer milliseconds; `floor` here, don't `round`!
+NOW = NOW.replace(microsecond=(NOW.microsecond // 1000 * 1000))
+
 
 def test_qtmsghandler():
     """Test qtpy.QtMsgHandler"""
     assert QtCore.qInstallMessageHandler is not None
 
 
-def test_qdatetime_toPython():
-    """Test QDateTime.toPython"""
-    q_date = QtCore.QDateTime.currentDateTime()
-    assert QtCore.QDateTime.toPython is not None
-    py_date = q_date.toPython()
-    assert isinstance(py_date, datetime)
+def test_QDateTime_toPython_and_toPyDateTime():
+    """Test `QDateTime.toPython` and `QDateTime.toPyDateTime`"""
+    q_datetime = QtCore.QDateTime(NOW)
+    py_date = q_datetime.toPython()
+    assert py_date == NOW
+    py_date = q_datetime.toPyDateTime()
+    assert py_date == NOW
```

Comment:
I suggest paratermizing each of the tests for the two methods instead of conflating the two and duplicating logic, with the name of the method to call, i.e. something like

```suggestion
@pytest.mark.parametrize("method", ["toPython", "toPyDateTime"])
def test_QDateTime_toPython_and_toPyDateTime(method):
    """Test `QDateTime.toPython` and `QDateTime.toPyDateTime`"""
    q_datetime = QtCore.QDateTime(NOW)
    py_date = getattr(q_datetime, method)()
    assert py_date == NOW
```

And the same for the other tests.

Reply:
Done: https://github.com/spyder-ide/qtpy/pull/421/commits/553338d443794c662bbe4a8ae8e287cf463535fe